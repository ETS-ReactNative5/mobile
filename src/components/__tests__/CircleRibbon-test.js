import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { CircleRibbon } from '../CircleRibbon'

it('renders correctly with projects', () => {
  let projects = {
    '11':
     { name: 'Whales as Individuals',
       slug: 'tedcheese/whales-as-individuals',
       activity_count: 5,
       sort_order: 3 },
    '14':
     { name: 'Snapshots at Sea',
       slug: 'tedcheese/snapshots-at-sea',
       activity_count: 7,
       sort_order: 2 },
    '17':
     { name: 'Planet Four: Terrains',
       slug: 'mschwamb/planet-four-terrains',
       activity_count: 7,
       sort_order: 1 }
  }

  let totalClassifications = 19

  const tree = renderer.create(
    <CircleRibbon projects={projects} totalClassifications={totalClassifications} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with a majority project', () => {
  let projects = {
    '11':
     { name: 'Whales as Individuals',
       slug: 'tedcheese/whales-as-individuals',
       activity_count: 5,
       sort_order: 3 },
    '14':
     { name: 'Snapshots at Sea',
       slug: 'tedcheese/snapshots-at-sea',
       activity_count: 7,
       sort_order: 2 },
    '17':
     { name: 'Planet Four: Terrains',
       slug: 'mschwamb/planet-four-terrains',
       activity_count: 70,
       sort_order: 1 }
  }

  let totalClassifications = 82

  const tree = renderer.create(
    <CircleRibbon projects={projects} totalClassifications={totalClassifications} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with 1 project', () => {
  let projects = {
    '11':
     { name: 'Whales as Individuals',
       slug: 'tedcheese/whales-as-individuals',
       activity_count: 5,
       sort_order: 1 }
  }

  let totalClassifications = 5

  const tree = renderer.create(
    <CircleRibbon projects={projects} totalClassifications={totalClassifications} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with no projects', () => {
  let projects = {}
  let totalClassifications = null

  const tree = renderer.create(
    <CircleRibbon projects={projects} totalClassifications={totalClassifications} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with 1 classification', () => {
  let projects = {
    '11':
     { name: 'Whales as Individuals',
       slug: 'tedcheese/whales-as-individuals',
       activity_count: 1,
       sort_order: 1 }
    }
  let totalClassifications = 1

  const tree = renderer.create(
    <CircleRibbon projects={projects} totalClassifications={totalClassifications} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
