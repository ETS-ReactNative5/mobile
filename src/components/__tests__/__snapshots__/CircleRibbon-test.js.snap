// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with 1 classification 1`] = `
<View>
  <Svg
    height={120}
    width={120}
  >
    <Path
      d="
            M 60 10
            A 50 50 0 1 1 59.99999999999999, 10
          "
      fill="transparent"
      stroke="hsl(277, 57%, 50%)"
      strokeWidth={10}
    />
  </Svg>
</View>
`;

exports[`renders correctly with 1 project 1`] = `
<View>
  <Svg
    height={120}
    width={120}
  >
    <Path
      d="
            M 60 10
            A 50 50 0 1 1 59.99999999999999, 10
          "
      fill="transparent"
      stroke="hsl(277, 57%, 50%)"
      strokeWidth={10}
    />
  </Svg>
</View>
`;

exports[`renders correctly with a majority project 1`] = `
<View>
  <Svg
    height={120}
    width={120}
  >
    <Path
      d="
            M 60 10
            A 50 50 0 1 1 20.23535643632868, 29.688729451668095
          "
      fill="transparent"
      stroke="hsl(189, 79%, 50%)"
      strokeWidth={10}
    />
    <Path
      d="
            M 20.23535643632868 29.688729451668095
            A 50 50 0 0 1 41.309146407961514, 13.624877444895283
          "
      fill="transparent"
      stroke="hsl(22, 82%, 50%)"
      strokeWidth={10}
    />
    <Path
      d="
            M 41.309146407961514 13.624877444895283
            A 50 50 0 0 1 59.99999999999999, 10
          "
      fill="transparent"
      stroke="hsl(277, 57%, 50%)"
      strokeWidth={10}
    />
  </Svg>
</View>
`;

exports[`renders correctly with no projects 1`] = `
<View>
  <Svg
    height={120}
    width={120}
  />
</View>
`;

exports[`renders correctly with projects 1`] = `
<View>
  <Svg
    height={120}
    width={120}
  >
    <Path
      d="
            M 60 10
            A 50 50 0 0 1 96.78619553365658, 93.86407858128705
          "
      fill="transparent"
      stroke="hsl(189, 79%, 50%)"
      strokeWidth={10}
    />
    <Path
      d="
            M 96.78619553365658 93.86407858128705
            A 50 50 0 0 1 10.170775349666506, 64.12896727361664
          "
      fill="transparent"
      stroke="hsl(22, 82%, 50%)"
      strokeWidth={10}
    />
    <Path
      d="
            M 10.170775349666506 64.12896727361664
            A 50 50 0 0 1 59.99999999999999, 10
          "
      fill="transparent"
      stroke="hsl(277, 57%, 50%)"
      strokeWidth={10}
    />
  </Svg>
</View>
`;
